src/main/java/Duke.java

```diff
@@ -1,10 +1,380 @@
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Scanner;
+import java.io.File;
+import java.io.FileNotFoundException;
+
+
 public class Duke {
-    public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+
+    //taskList variable for level-2 onwards
+    private static ArrayList<Task> taskList = new ArrayList<Task>();
+    //task count
+    private static int taskCount = 0;
+
+    private static String folderPath = "./data";
+    private static String filePath = folderPath+ "/duke.txt";
+
+
+    //echo function for level-1
+    public static void echo(String message) {
+        System.out.println(message);
+    }
+
+
+    //add function for level-2
+    public static void add(String task) {
+        Task newTask = new Task(task);
+        taskList.add(newTask);
+        taskCount++;
+        System.out.println("added: " + task);
+    }
+
+
+    //list function for level-2
+    public static void list() {
+        if(taskCount == 0) {
+            System.out.println("No tasks found");
+        }
+        else {
```

Comment:
Readability can be improved here by swapping the else-clause with the if-clause.